File: services/forum/src/main/java/com/capfest/forum/entity/Discussion.java
package com.capfest.forum.entity;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;


import java.util.Date;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "discussions")
public class Discussion {
    @Id
    private String discussionId;
    private String title;
    private String content;
    private Date createdAt;
    private String userId;
    private VisibilityType visibility;
    private int likes;
    private int comments;
    private List<String> tags;
    private boolean pinned;
    private int viewCount;
    private double sentimentScore;
    private List<Comment> commentList;
    private List<String> invitedUsers;
    private DiscussionStatus status;
    private Set<String> likedByUserIds = new HashSet<>(); // New field to track user likes


}

File: src/model/Forum/Discussion.ts
import {Comment} from "./Comment";

export interface Discussion {
  discussionId: string;
  title: string;
  content: string;
  createdAt: Date;
  userId: string;
  visibility: VisibilityType;
  likes: number;
  comments: number;
  tags: string[];
  pinned: boolean;
  viewCount: number;
  sentimentScore: number;
  commentList: Comment[]; // Ensure this is correctly defined as an array of Comment objects
  invitedUsers: string[];
  status: DiscussionStatus;
  userHasLiked: boolean; // Add this line

}
export enum VisibilityType {
  PUBLIC = 'PUBLIC',
  PRIVATE = 'PRIVATE',
  INVITE_ONLY = 'INVITE_ONLY'
}

export enum DiscussionStatus {
  PENDING_APPROVAL = 'PENDING_APPROVAL',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED'
}

